test/index.js

```diff
@@ -251,13 +252,89 @@ describe('Intercom', function(){
   })
 
   describe('.track()', function(){
-    it('should track', function(done){
-      var json = test.fixture('track-basic');
+    // Create one persistent unique user to properly test locking logic
+    // Also declare jobId for later tests
+    var userId;
+    var jobId;
+    before(function(){
+      userId = uid();
+    });
+
+    beforeEach(function(done){
+      // Make a quick `.identify()` call with the userId so Intercom will
+      // accept these `.track()` calls
+      var userRequest = {
+        type: 'identify',
+        timestamp: '2016',
+        userId: userId,
+        traits: { created: '2016' }
+      };
+      test
+        .request(0)
+        .set(settings)
+        .identify(userRequest)
+        .expects(200)
+        .end(done);
+    });
+
+    it('should create new job for track', function(done){
+      var json = test.fixture('track-job-new');
+      json.input.userId = userId;
+      json.output.items[0].data.user_id = userId;
+
       test
+        .request(1) // second req after beforeEach
         .set(settings)
         .track(json.input)
         .sends(json.output)
-        .expects(202)
+        .expects(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          // save jobId for next test
+          jobId = res[0].res.body.id;
```

Comment:
This seems sketchy. We shouldn't really rely on test ordering for them to pass. If there's a dependency, run it within the same test or in a beforeEach.